lang/csharp/src/apache/main/POCO/POCOWriter.cs

```diff
@@ -0,0 +1,250 @@
+    /// </summary>
+    /// <typeparam name="T">type name of specific object</typeparam>
+    public class POCOWriter<T> : SpecificWriter<T>
+    {
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="schema"></param>
+        /// <returns></returns>
+        public POCOWriter(Schema schema) : base(new POCODefaultWriter(typeof(T), schema)) { }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="writer"></param>
+        /// <returns></returns>
+        public POCOWriter(POCODefaultWriter writer) : base(writer) { }
+    }
+    /// <summary>
+    /// Class for writing data from any specific objects
+    /// </summary>
+    public class POCODefaultWriter : SpecificDefaultWriter
+    {
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="objType"></param>
+        /// <param name="schema"></param>
+        public POCODefaultWriter(Type objType, Schema schema)
+            : base(schema)
+        {
+            var rs = schema as RecordSchema;
+            if (rs != null)
+            {
+                ClassCache.LoadClassCache(objType, rs);
+            }
+        }
+
+        /// <summary>
+        /// Serialized a record using the given RecordSchema. It uses GetField method
+        /// to extract the field value from the given object.
+        /// </summary>
+        /// <param name="schema">The RecordSchema to use for serialization</param>
+        /// <param name="value">The value to be serialized</param>
+        /// <param name="encoder">The Encoder for serialization</param>
+
+        protected override void WriteRecord(RecordSchema schema, object value, Encoder encoder)
+        {
+
+            foreach (Field field in schema)
+            {
+                try
+                {
+                    var v = ClassCache.GetClass(schema).GetValue(value, field);
+
+                    Write(field.Schema, v, encoder);
+                }
+                catch (Exception ex)
+                {
+                    throw new AvroException(ex.Message + " in field " + field.Name, ex);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Validates that the record is a fixed record object and that the schema in the object is the
+        /// same as the given writer schema. Writes the given fixed record into the given encoder
+        /// </summary>
+        /// <param name="schema">writer schema</param>
+        /// <param name="value">fixed object to write</param>
+        /// <param name="encoder">encoder to write to</param>
+        protected override void WriteFixed(FixedSchema schema, object value, Encoder encoder)
+        {
+            var fixedrec = value as byte[];
+            if (fixedrec == null)
+            {
+                throw new AvroTypeException("Fixed object is not derived from byte[]");
+            }
```

Comment:
I think it would be nice to also allow derivatives of `GenericFixed` as `value` in this method. This would allow users of the library to use some of their auto-generated Specific API classes in the POCO API.